contracts/sidechain/AuraBalProxyOFT.sol

```diff
@@ -0,0 +1,399 @@
+
+        claimable[_token][_srcChainId] = 0;
+        totalClaimable[_token] -= reward;
+
+        if (_token == address(innerToken)) {
+            require(oft == address(this), "!oft");
+            // The token is this inner token so we need to call the internal
+            // bridge mint/burn functions rather than sendFrom
+            internalTotalSupply += reward;
+            _lzSend(
+                _srcChainId,
+                abi.encode(PT_SEND, abi.encodePacked(receiver), reward),
+                payable(msg.sender),
+                configs[_srcChainId][AuraBalProxyOFT.processClaimable.selector].zroPaymentAddress,
+                configs[_srcChainId][AuraBalProxyOFT.processClaimable.selector].adapterParams,
+                msg.value
+            );
+
+            emit SendToChain(_srcChainId, address(this), abi.encode(receiver), reward);
+        } else {
+            // The token is one that this contract holds a balance of eg $AURA
+            // bridge it to the L2 via it's proxy OFT contracts
+            IERC20(_token).safeIncreaseAllowance(oft, reward);
+            IProxyOFT(oft).sendFrom{ value: msg.value }(
+                address(this),
+                _srcChainId,
+                abi.encodePacked(receiver),
+                reward,
+                payable(msg.sender),
+                configs[_srcChainId][AuraBalProxyOFT.processClaimable.selector].zroPaymentAddress,
+                configs[_srcChainId][AuraBalProxyOFT.processClaimable.selector].adapterParams
+            );
+        }
+    }
+
+    /* -------------------------------------------------------------------
+      Vault Wrapper 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @notice Execute a function on the vault
+     * @dev    In order to account for the withdrawalPenalty this contract needs
+     *         To be the owner of the auraBAL vault. Therefore it needs to be
+     *         able to call vault owner functions. Rather than wrapping each
+     *         function we can just use an execute pointing at the vault
+     */
+    function vaultExecute(uint256 _value, bytes calldata _data) external onlyOwner returns (bool, bytes memory) {
+        (bool success, bytes memory result) = vault.call{ value: _value }(_data);
+        require(success, "!success");
+
+        return (success, result);
+    }
+
+    /* -------------------------------------------------------------------
+      Overrides 
+    ------------------------------------------------------------------- */
+
+    function rescue(
+        address _token,
+        address _to,
+        uint256 _amount
+    ) external override {
+        require(msg.sender == sudo, "!sudo");
+
+        // Adjust the internalTotalSupply. This means we have to harvest and process
+        // any rewards if we want to rescue the entire underlyingBalance of the bridge
+        // otherwise this will underflow
+        internalTotalSupply -= _amount;
```

Comment:
Should not this line     `internalTotalSupply -= _amount;` be inside  the if ?

```
       if (_token == address(innerToken)) {
          internalTotalSupply -= _amount;
            _withdraw(_amount);
        }
 ```

Reply:
yep. In fact I'm have tweaked this to not accept a _token address. I have a change in my local branch for this. Will raise a PR shortly